db/users.js

```diff
@@ -0,0 +1,126 @@
+const client = require("./client");
+const bcrypt = require("bcrypt");
+
+// createUser({ username, password })
+// hash the password before storing it to the database
+async function createUser({ username, password }) {
+  try {
+    const hashedPassword = await bcrypt.hash(password, 10);
+
+    const {
+      rows: [user],
+    } = await client.query(
+      `
+              INSERT INTO users(username, email, password) 
+              VALUES($1, $2) 
+              ON CONFLICT (username) DO NOTHING
+              RETURNING *;
+          `,
+      [username, hashedPassword]
+    );
+
+    delete user.password;
+    return user;
+  } catch (error) {
+    throw error;
+  }
+}
+
+// getUser({ username, password })
+// verify the password against the hashed password
+async function getUser({ username, password }) {
+  try {
+    const user = await getUserByUsername(username);
+    const hashedPassword = user.password;
```

Comment:
We could also add more error handling which will be helpful later on like this: 

if (!user) {
     throw {
          name: "UserNotFound",
          message: "User not found!" or whatever message that can help us on the frontend
     }
}